operator/src/main/java/oracle/kubernetes/operator/DomainRecheck.java

```diff
@@ -207,12 +210,24 @@ Step startNamespaceSteps(String ns) {
     @Override
     public NextAction apply(Packet packet) {
       NamespaceStatus nss = domainNamespaces.getNamespaceStatus(ns);
-      if (fullRecheck || !nss.isNamespaceStarting().getAndSet(true)) {
+      if (fullRecheck) {
         return doNext(packet);
+      } else if (isNamespaceStartingChangedToTrue(nss)) {
+        return doNext(addNSWatchingStartingEventStep(), packet);
       } else {
         return doEnd(packet);
       }
     }
+
+    private Step addNSWatchingStartingEventStep() {
+      return Step.chain(
+          EventHelper.createEventStep(new EventData(NAMESPACE_WATCHING_STARTED).namespace(ns).resourceName(ns)),
+          getNext());
+    }
+
+    private boolean isNamespaceStartingChangedToTrue(NamespaceStatus nss) {
```

Comment:
The name of this method is problematic, as it explains how it is implemented, rather than what behavior it provides. I presume that the intent is that the namespace should be started? And that it only happens once?

If so, maybe something like, `shouldStartNamespace` would be better. Better still would be to put that method on `DomainNamespaces` in order to start hiding the implementation of the flag.

Reply:
Did you mean to move the method (after rename it) to DomainStatus (instead of DomainnNamespaces) class?

Reply:
I think I understand what you meant now. thanks.